lib/basedriver/capabilities.js

```diff
@@ -101,11 +98,20 @@ function stripAppiumPrefixes (caps) {
   if (badPrefixedCaps.length > 0) {
     throw new errors.InvalidArgumentError(`The capabilities ${JSON.stringify(badPrefixedCaps)} are standard capabilities and should not have the "appium:" prefix`);
   }
+}
 
-  // If client provides non-prefixed, non-standard capabilities, warn them that these should be prefixed
-  if (unprefixedCaps.length > 0) {
-    log.warn(`${JSON.stringify(unprefixedCaps)} are not standard capabilities and should have an extension prefix`);
-  }
+/**
+ * Get an array of all the unprefixed caps that are being used in 'alwaysMatch' and all of the 'firstMatch' object
+ * @param {Object} caps A capabilities object
+ */
+function findNonPrefixedCaps ({alwaysMatch={}, firstMatch=[]}) {
+  return _.chain([alwaysMatch, ...firstMatch])
+    .reduce((unprefixedCaps, caps) => [
+      ...unprefixedCaps,
+      ..._(caps).keys().filter((cap) => !cap.includes(':') && !isStandardCap(cap)),
```

Comment:
what `_(caps)` is doing?

Reply:
will it work if cap == null?

Reply:
https://lodash.com/docs/4.17.4#lodash

Reply:
@mykola-mokhnach _(caps) allows you to chain together lodash functions to avoid callback messes. And I don't think `cap` could be null because `.keys()` returns a list of property names.